### DISCUSSION TITLE
wxWidgets webview not building on linux

### DISCUSSION THREAD

**Discussion by Jono-Moss** [microsoft/vcpkg#27997] [Q&A]
when using the following on linux (pop-os):
`vcpkg install wxwidgets[webview]`
I get the following error
```
jono@pop-os:~/DataHdd/SDKs/vcpkg$ ./vcpkg install wxwidgets[webview]:x64-linux
Computing installation plan...
webview2[core]:x64-linux is only supported on 'windows & (x86 | x64 | arm64)'
```
how would i go about telling vcpkg to use normal webview (wxWEBVIEW_WEBKIT (GTK)) see below:
[https://docs.wxwidgets.org/3.0/classwx_web_view.html](https://docs.wxwidgets.org/3.0/classwx_web_view.html)
and not webview2 (edge) engine

**Comment by Jono-Moss** [comment]
Okay I seem to have solved it by using the following:
`vcpkg install wxwidgets[webview] --allow-unsupported --recurse`

**Reply by dg0yt** [reply]
This is a workaround, not a solution. vcpkg doesn't provide the webkitgtk lib. So your build depends on system libraries - and they may pull in other system libraries which could eventually clash (at runtime) with libraries which vcpkg gtk takes from vcpkg. YMMV.

**Reply by Jono-Moss** [reply]
Thank you, I have unmarked it as a solution. 
I see what you mean. When I run my application it crashes when trying to use the webview. 
Will it be possible for me to build wxWidgets webview separately and then copy the libraries to the vcpkg install directory. 
I have tried it that way but with no luck so far. I don't know if I'm supposed to tell vcpkg that i have added it some how.
I initially built wxWidgets 3.2 from source,  but no matter what I do I can't seem to get cLion / cMake to find it. 
The only why I can get cMake to find wxWidgets is via vcpkg.

I have tried many solutions such as:
https://forums.wxwidgets.org/viewtopic.php?p=197038#p187276
but wxWidgets 3.2 just doesn't get found.

**Reply by dg0yt** [reply]
wxWidget now has now exported CMake config, so this should work to some extent even without vcpkg

For the vcpkg port, I spend some time on the wrapper for CMake's `FindwxWidget.cmake`, and I added test ports for CI.

> Will it be possible for me to build wxWidgets webview separately and then copy the libraries to the vcpkg install directory.

Ideally, you would create the missing webkit2gtk port for vcpkg. But it is not trivial. Dependencies must be fed into the build, and they must be appear correctly in exported configuration. And there are many of them. See the build dependencies in Debian: 
https://sources.debian.org/src/webkit2gtk/2.38.2-1/debian/control/

**Reply by Jono-Moss** [reply]
Thank you very much for all your help. 
I'm still trying to wrap my head around cMake getting it to find packages cross platform, so I think I won't have the ability to build a  webkit2gtk port yet :). 
Thanks again for all the help. 
I'll continue to try get the source build to work with cMake, if I'm still struggling after a few days all attempt the port.
